In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
girls12_df = pd.read_csv("/content/drive/MyDrive/Data Science/Heights_Girls12_Cleaned.csv")

In [12]:
girls12_df.head()

,Sex,Year,Age group,Mean height,Mean height lower 95% uncertainty interval,Mean height upper 95% uncertainty interval,Mean height standard error
0,Girls,1985,12,145.797559,145.536314,146.056138,0.131974
1,Girls,1986,12,145.822902,145.571653,146.068954,0.126969
2,Girls,1987,12,145.836667,145.594924,146.078006,0.122385
3,Girls,1988,12,145.842282,145.609922,146.072701,0.118209
4,Girls,1989,12,145.844329,145.617697,146.068599,0.114863


In [4]:
from scipy.stats import linregress
result = linregress(girls12_df['Year'],girls12_df['Mean height'])
result

LinregressResult(slope=np.float64(0.08348932230598419), intercept=np.float64(-20.384737703680912), rvalue=np.float64(0.9641797303156522), pvalue=np.float64(1.3643234731964378e-20), stderr=np.float64(0.0039982597669712665), intercept_stderr=np.float64(8.00461790659875))

Slope: 0.0835

p-value: 1.3643 e-20

There is statistically significant evidence of a relationship between year and mean height for 12-year-old girls from 1985 to 2019, so we reject the null hypothesis of no change.

In [8]:
import numpy as np
boot_slopes = []
for _ in range(5000):
  sample = girls12_df.sample(n=len(girls12_df), replace=True)
  boot_slopes.append(linregress(sample['Year'], sample['Mean height']).slope)
np.percentile(boot_slopes, [2.5, 97.5])

array([0.07267367, 0.09418227])

The estimated yearly increase is plausibly between about 0.0727 and 0.0942 cm per year, and that interval doesn't include 0, which points to an upward trend.

- confounder: average childhood nutrition quality over time
 - subgroup check: early years vs recent years

In [11]:
early = girls12_df[girls12_df['Year'] < 2002]
recent = girls12_df[girls12_df['Year'] >= 2002]
early.shape, recent.shape
early['Mean height'].mean(), recent['Mean height'].mean()

(np.float64(146.0152408545164), np.float64(147.46510554581667))

Null case sentence: I found no significant relationship between the year and mean height for 12-year-old girls, and my data could not distinguish a small trend from no trend.